projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -467,18 +467,12 @@ public static function user_can_edit() {
 	 * @return bool Whether the post can be viewed
 	 */
 	public static function user_can_view_post() {
-		$user_id = get_current_user_id();
-		$post_id = get_the_ID();
-
-		if ( empty( $post_id ) ) {
-			// We are not viewing a post.
-			$post_id = 0;
-		}
-
+		$user_id   = get_current_user_id();
+		$post_id   = get_the_ID();
 		$cache_key = sprintf( '%d_%d', $user_id, $post_id );
 
-		if ( isset( self::$user_can_view_post_cache[ $cache_key ] ) ) {
```

Comment:
Be careful. I don't think you want to remove the cache

Reply:
Fixed in [8cc6887](https://github.com/Automattic/jetpack/pull/31254/commits/8cc68876ef253f94718ce372efd5208756ebdf8e)